src/index.js

```diff
@@ -177,48 +180,101 @@ class Saturn {
   }
 
   async _reportLogs () {
-    if (this.logs.length) {
-      await fetch(
-        'https://twb3qukm2i654i3tnvx36char40aymqq.lambda-url.us-west-2.on.aws/',
-        {
-          method: 'POST',
-          body: JSON.stringify({
-            bandwidthLogs: this.logs
-          })
-        }
-      )
-      this.logs = []
+    if (!this.logs.length) {
+      return
+    }
+
+    let bandwidthLogs = this.logs
+    if (this.hasPerformanceAPI) {
+      bandwidthLogs = this._matchLogsWithPerformanceMetrics(bandwidthLogs)
     }
+
+    await fetch(
+      this.opts.logURL,
+      {
+        method: 'POST',
+        body: JSON.stringify({ bandwidthLogs, logSender: this.opts.logSender })
+      }
+    )
+
+    this.logs = []
+    this._clearPerformanceBuffer()
+  }
+
+  /**
+   *
+   * @param {Array<object>} logs
+   */
+  _matchLogsWithPerformanceMetrics (logs) {
+    const logsToReport = []
+
+    for (const log of logs) {
+      const metrics = this._getPerformanceMetricsForLog(log)
+
+      if (!metrics.isFromBrowserCache) {
+        delete metrics.isFromBrowserCache
+        Object.assign(log, metrics)
+        logsToReport.push(log)
+      }
+    }
+
+    return logsToReport
   }
 
   /**
    *
    * @param {object} log
+   * @returns {object}
    */
-  _addPerformanceAPIMetrics (log) {
-    if (typeof window !== 'undefined' && window?.performance) {
-      const entry = performance
-        .getEntriesByType('resource')
-        .find((r) => r.name === log.url.href)
-      if (entry) {
-        const dnsStart = entry.domainLookupStart
-        const dnsEnd = entry.domainLookupEnd
-        const hasData = dnsEnd > 0 && dnsStart > 0
-        if (hasData) {
-          log.dnsTimeMs = Math.round(dnsEnd - dnsStart)
-          log.ttfbAfterDnsMs = Math.round(
-            entry.responseStart - entry.requestStart
-          )
-        }
-
-        if (log.httpProtocol === null && entry.nextHopProtocol) {
-          log.httpProtocol = entry.nextHopProtocol
-        }
-        // else response didn't have Timing-Allow-Origin: *
-        //
-        // if both dnsStart and dnsEnd are > 0 but have the same value,
-        // its a dns cache hit.
+  _getPerformanceMetricsForLog (log) {
+    const metrics = {}
+
+    const entry = performance
+      .getEntriesByType('resource')
+      .find((r) => r.name === log.url.href)
+
+    if (entry) {
+      const dnsStart = entry.domainLookupStart
+      const dnsEnd = entry.domainLookupEnd
+      const hasDnsMetrics = dnsEnd > 0 && dnsStart > 0
+
+      if (hasDnsMetrics) {
+        metrics.dnsTimeMs = Math.round(dnsEnd - dnsStart)
```

Comment:
why does this need to be rounded?

Reply:
same below

Reply:
The dns values are floats with microsecond precision, I dont think we need that much precision and the db column type is integer.

Reply:
Thanks, I didn't know those were floats